lib/modules/asset/AssetService.ts

```diff
@@ -135,6 +135,97 @@ export class AssetService extends BaseService {
     });
   }
 
+  /**
+   * Update or Create an asset metadata
+   */
+  /**
+   * Update or Create an asset metadata
+   */
+  public async upsert(
+    engineId: string,
+    assetId: string,
+    metadata: Metadata,
+    request: KuzzleRequest
+  ): Promise<KDocument<AssetContent>> {
+    return lock(`asset:${engineId}:${assetId}`, async () => {
+      let asset = null;
+      try {
+        asset = await this.get(engineId, assetId, request);
+      } catch (error) {
+        this.app.log.error(`Asset ${assetId} not found`);
+        asset = {
+          _id: assetId,
+          _source: {
+            groups: [],
+            lastMeasuredAt: 0,
+            linkedDevices: [],
+            measures: undefined,
+            metadata: {},
+            model: "",
+            reference: "",
+          },
+        };
+      }
+
+      const upsertedPayload = await this.app.trigger<EventAssetUpdateBefore>(
+        "device-manager:asset:update:before",
```

Comment:
Should be "device-manager:asset:upsert:before" .  Not sure there is corresponding event type, in this case we should prob add it.

Reply:
This is marked as resolved but this is still update event, it should be upsert